protocol/x/listing/keeper/listing_test.go

```diff
@@ -265,27 +278,129 @@ func TestCreateClobPair(t *testing.T) {
-						clobPair.GetClobPairSubticksPerTick(),
-					)
-					require.Equal(
-						t,
-						types.DefaultStepBaseQuantums,
-						clobPair.GetClobPairMinOrderBaseQuantums().ToUint64(),
-					)
-					require.Equal(t, perpetualId, clobPair.MustGetPerpetualId())
 				} else {
-					_, found := clobKeeper.GetClobPair(ctx, clobtypes.ClobPairId(clobPairId))
 					require.False(t, found)
 				}
 			},
 		)
 	}
 }
+
+func TestDepositToMegavaultforPML(t *testing.T) {
+	tests := map[string]struct {
+		address    string
+		balance    *big.Int
+		asset      asstypes.Asset
+		clobPairId uint32
+
+		expectedErr string
+	}{
+		"success": {
+			address:    constants.AliceAccAddress.String(),
+			balance:    big.NewInt(20_000),
+			asset:      *constants.Usdc,
+			clobPairId: 1,
+
+			expectedErr: "",
+		},
+		"failure - insufficient balance": {
+			address:    constants.AliceAccAddress.String(),
+			balance:    big.NewInt(0),
+			asset:      *constants.Usdc,
+			clobPairId: 1,
+
+			expectedErr: "NewlyUndercollateralized",
+		},
+		"failure - invalid clob pair id": {
+			address:    constants.AliceAccAddress.String(),
+			balance:    big.NewInt(20_000),
+			asset:      *constants.Usdc,
+			clobPairId: 100, // non existent clob pair id
+
+			expectedErr: "ClobPair not found",
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(
+			name, func(t *testing.T) {
+				tApp := testapp.NewTestAppBuilder(t).WithGenesisDocFn(
+					func() (genesis comettypes.GenesisDoc) {
+						genesis = testapp.DefaultGenesis()
+						// Initialize vault with its existing equity.
+						testapp.UpdateGenesisDocWithAppStateForModule(
+							&genesis,
+							func(genesisState *satypes.GenesisState) {
+								genesisState.Subaccounts = []satypes.Subaccount{
+									{
+										Id: &vaulttypes.MegavaultMainSubaccount,
+										AssetPositions: []*satypes.AssetPosition{
+											testutil.CreateSingleAssetPosition(
+												0,
+												big.NewInt(1_000_000),
+											),
+										},
+									},
+									{
+										Id: &satypes.SubaccountId{
+											Owner:  tc.address,
+											Number: 0,
+										},
+										AssetPositions: []*satypes.AssetPosition{
+											testutil.CreateSingleAssetPosition(
+												tc.asset.Id,
+												tc.balance,
+											),
+										},
+									},
+								}
+							},
+						)
+						return genesis
+					},
+				).Build()
+				ctx := tApp.InitChain()
+
+				// Set existing total shares.
+				err := tApp.App.VaultKeeper.SetTotalShares(
+					ctx,
+					vaulttypes.BigIntToNumShares(big.NewInt(1_000_000)),
+				)
+				require.NoError(t, err)
+
+				err = tApp.App.ListingKeeper.DepositToMegavaultforPML(
+					ctx,
+					satypes.SubaccountId{
+						Owner:  tc.address,
+						Number: 0,
+					},
+					tc.clobPairId,
+				)
+				if tc.expectedErr != "" {
+					require.ErrorContains(t, err, tc.expectedErr)
+				} else {
+					require.NoError(t, err)
+				}
+			},
+		)
```

Comment:
maybe also check vault state?

Reply:
Could have an E2E test that also tests that there exists an order for the vault after the next block.